os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -0,0 +1,826 @@
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_dh_compute_shared_secret(dh_param, dh_idx, shared_secret);
+	up_free_secure_context();
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_ecdh_compute_shared_secret(hal_ecdh_data *ecdh_param, uint32_t key_idx, hal_data *shared_secret)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if (key_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_ecdh_compute_shared_secret(ecdh_param, key_idx, shared_secret);
+	up_free_secure_context();
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_set_certificate(uint32_t cert_idx, hal_data *cert_in)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if (cert_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_set_certificate(cert_idx, cert_in);
+	up_free_secure_context();
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_get_certificate(uint32_t cert_idx, hal_data *cert_out)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if (cert_idx == FACTORYKEY_RTL_CERT) {	/* If factory Cert */
+		ret = ameba_hal_get_factory_cert(cert_idx, cert_out);
+		goto exit;
+	}
+
+	if (cert_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_get_certificate(cert_idx, cert_out);
+	up_free_secure_context();
+
+exit:
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_remove_certificate(uint32_t cert_idx)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	if (cert_idx >= KEY_STORAGE_INDEX_MAX) {	/* Index Out of Range */
+		return HAL_INVALID_SLOT_RANGE;
+	}
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_remove_certificate(cert_idx);
+	up_free_secure_context();
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_get_factory_key(uint32_t key_idx, hal_data *key)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_get_factory_key(key_idx, key);
+	up_free_secure_context();
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_get_factory_cert(uint32_t cert_idx, hal_data *cert)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_get_factory_cert(cert_idx, cert);
+	up_free_secure_context();
+
+	if (ret != HAL_SUCCESS) {
+		sedbg("RTL SE failed (%zu)\n", ret);
+	}
+	return ret;
+}
+
+int se_ameba_hal_get_factory_data(uint32_t data_idx, hal_data *data)
+{
+	AWRAP_ENTER;
+	int ret;
+
+	up_allocate_secure_context(4096);
+	ret = ameba_hal_get_factory_data(data_idx, data);
```

Comment:
Is it okay not to check that `data_idx` is in valid range or not?

Reply:
Factory_key index is a fix index, not range, the index is pass into SE image3.
This index will be check inside the SE image3.